### ISSUE TITLE
segfault when incorrect column types are provided and fill=TRUE

### ISSUE THREAD

**Issue by anasem** [Rdatatable/data.table#4525]
The following example produces a segfault when read with `colClasses = list("numeric" =1:2))` 
Note the provided string has a character on line 6

`#` [`Minimal reproducible example`](https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example)

```R
txt <- "1,1
10,10
20,20
3,3
10,5
r,8
17,18
"


dt <- fread(text=txt, fill = TRUE, colClasses = list("numeric" =1:2))

```

`#` `Output of sessionInfo()`
```

R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Manjaro Linux

Matrix products: default
BLAS:   /usr/lib/libblas.so.3.9.0
LAPACK: /usr/lib/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] data.table_1.12.9 colorout_1.2-2   

loaded via a namespace (and not attached):
[1] compiler_4.0.0
```

**Comment by jangorecki** [comment]
Thank you for reporting. Reproducible on 1.12.8 as well.

**Comment by MichaelChirico** [comment]
Not the same as #4337 right?

**Comment by ben-schwen** [comment]
> Not the same as #4337 right?

I would say its the same as #4337 

Segfault also occurs with 
`dt = fread(text=txt, select=1L, colClasses = "integer")` 
and only with a higher version than `1.12.2`

**Comment by tlapak** [comment]
This and #4337 (which is indeed the same) were duplicated by #4644 which in turn was closed by #4724. I've confirmed that both no longer crash as of 1.13.2 but did crash on 1.13.0. I'll close both issues.
@mattdowle should the news item be amended to reference these reports?

**Comment by mattdowle** [comment]
Thanks for investigating and updating the status.
I haven't looked at these particular ones, but here's how I currently think about such things.

1. The first/earliest report should be credited. So if that person is not already thanked, then yes the news item should be updated to reference the first report and thank them.

2. At the time of fixing, we often thank the people filing duplicates too. They often have spent time producing good reports and there is often a nuance making it not an exact duplicate. Tests are often added verbatim from all the duplicates just because it doesn't take much extra time to do so and different angles of testing can sometimes reveal unexpected nuances.

3. But if time has passed since the fix, and duplicates weren't properly closed down at the time, then I would just close the duplicate making sure to link to the fix, the first report, and paste an image of the news item that fixed it. PRs (even just for a news item) take time, and although we probably would have thanked the person if we'd linked the issue at the time, that's an inconsistency we live with for the benefit of making progress given that we have a long backlog.

4. There have been a few cases of internal rework fixing cases we didn't look at at the time. Jan has been adding the tests from these reports and I've been double-checking which version fixed them.  In those cases we have been adding a news item thanking the reporter.  That's quicker though because we're already in a branch and PR to add the test, so added a news item is less work on top.
